File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        dfs(root,res);
        return res;
    }

    private void dfs(TreeNode root, List<Integer> tree){
        if (null!=root){
            if (null!=root.left){
                dfs(root.left,tree);
            }
            tree.add(root.val);
            if (null!=root.right){
                dfs(root.right,tree);
            }
        }
    }
}

File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        return self.dfs(root,[])
    
    def dfs(self, root, res):
        if root:
            if root.left:
                res = self.dfs(root.left,res)
            res.append(root.val)
            if root.right:
                res = self.dfs(root.right,res)
        return res
        